ts/components/ui/MultiImage.tsx

```diff
@@ -36,7 +42,30 @@ export class MultiImage extends React.PureComponent<Props, State> {
     if (sourceIndex === undefined) {
       return null;
     }
-    const source = this.props.source[sourceIndex];
+
+    const atIndex = this.props.source[sourceIndex];
+
+    // Workaround for invalidating the Android chance of the Image component.
+    const source: ImageURISource | ImageRequireSource = pipe(
+      Platform.OS === "android",
+      B.fold(
+        () => atIndex,
+        () =>
+          pipe(
+            atIndex,
+            T.number.decode,
+            E.fold(
+              () => {
+                const { uri, ...others } = atIndex as ImageURISource;
+                const timestampValue = toAndroidCacheTimestamp();
+                return { uri: `${uri}?ts=${timestampValue}`, ...others };
+              },
```

Comment:
What do you think of this refactor? Do you mean that it might work?
```suggestion
              () => { uri: `${atIndex.uri}?ts=${toAndroidCacheTimestamp()}`, ... atIndex.others },
```

Reply:
I think we still need to cast ```atIndex as ImageURISource``` because TS cannot infer its type after folding the decode result, thus we cannot destructure it.
What do you think? Thanks for your input.